File: Solutions/1684/1684.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1684.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-11-08 10:10:20
***************************************************************************************************/

int countConsistentStrings(char * allowed, char ** words, int wordsSize){
    int ret = 0;
    for (int i=0; i<wordsSize; i++) {
        bool flag1 = true;
        for (int j=0; j<strlen(words[i]); j++) {
            bool flag2 = false;
            for (int k=0; k<strlen(allowed); k++) {
                if (allowed[k] == words[i][j]) {
                    flag2 = true;
                    break;
                }
            }
            if (!flag2) {
                flag1 = false;
                break;
            }
        }
        if (flag1) ret++;
    }
    return ret;
}

File: Solutions/1684/1684.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/11/8 10:10
# File: 1684.py
# Desc: 

class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        ret = 0
        s = set(allowed)
        for word in words:
            for j in set(word):
                if j not in s: break
            else: ret+=1
        return ret
